src/code/z_vr_box.c

```diff
@@ -126,43 +126,104 @@ s32 func_800ADBB0(SkyboxContext* skyboxCtx, Vtx* roomVtx, s32 arg2, s32 arg3, s3
     s32 sp1F0[9 * 5];
     s32 sp13C[9 * 5];
     s32 sp88[9 * 5];
+    s32 *temp358;
+    s32 *temp2A4;
+    s32 *temp1F0;
+    s32 *temp13C;
+    s32 *temp88;
+    s16 temp3;
 
     switch (arg8) {
         case 0:
         case 2:
-            for (i = 0; i < 9; i++) {
-                for (j = 0; j < 5; j++) {
-                    sp358[5 * i + j] = arg3 + j * arg6;
-                    sp2A4[5 * i + j] = arg4 + i * arg7;
-                    sp1F0[5 * i + j] = arg5;
-                    sp13C[5 * i + j] = D_8012AD20[j];
-                    sp88[5 * i + j] = D_8012AD2C[i];
+            temp358 = sp358;
+            temp2A4 = sp2A4;
+            temp1F0 = sp1F0;
+            temp13C = sp13C;
+            temp88 = sp88;
+            pad428 = arg4;
+
+            for (i = 0; temp88 < sp88 + 45; pad428 += arg7, i++) {
+                pad42C = arg3;
+                temp3 = D_8012AD2C[i];
+                for (j = 0; j < 5; j++, pad42C += arg6) {
+                    *(temp358++) = pad42C;
+                    *(temp2A4++) = pad428;
+                    *(temp1F0++) = arg5;
+                    *(temp13C++) = D_8012AD20[j];
+                    *(temp88++) = temp3;
+                    
                 }
             }
+            // for (i = 0; i < 5; i++) {
+            //     for (j = 0; j < 5; j++) {
+            //         sp358[i][j] = arg3 + j * arg6;
+            //         sp2A4[i][j] = arg4 + i * arg7;
+            //         sp1F0[i][j] = arg5;
+            //         sp13C[i][j] = D_8012AD20[j];
+            //         sp88[i][j] = D_8012AD2C[i];
+            //     }
+            // }
```

Comment:
this commented out code and the two other commented out loops look like they should be removed since they rely on more than what's commented out (eg 2d arrays, and idk why i only goes up to 5 instead of 9)

Reply:
I'm leaving it in because it's more clear about what the loop is actually doing. The improved matching code I have in there is pretty hard to read and it's entirely possible the correct code looks nothing like it. The relevant arrays are listed as being `[5 * 5]` and `[9 * 5]`, which should give a hint as to what the 2D array's dimensions would be.

Reply:
what about `for (i = 0; i < 5; i++)` using 5 instead of 9

Reply:
also it wouldn't hurt leaving a comment to explain the 2d arrays and about this code is equivalent but matches less well

Reply:
oh, right, that was a copy/paste error. Fixing.